pkg/identity/buildkite/principal.go

```diff
@@ -69,16 +111,23 @@ func (p jobPrincipal) Name(_ context.Context) string {
 }
 
 func (p jobPrincipal) Embed(_ context.Context, cert *x509.Certificate) error {
-	// Set SubjectAlternativeName to the pipeline URL on the certificate
-	parsed, err := url.Parse(p.url)
+	baseURL, err := url.Parse(p.url)
 	if err != nil {
 		return err
 	}
-	cert.URIs = []*url.URL{parsed}
+
+	pipelineUrl := baseURL.JoinPath(p.organizationSlug, p.pipelineSlug)
+	jobUrl := baseURL.JoinPath(p.organizationSlug, p.pipelineSlug, "builds", strconv.FormatInt(p.buildNumber, 10)+"#"+p.jobId)
+
+	// Set SubjectAlternativeName to the pipeline URL on the certificate
+	cert.URIs = []*url.URL{pipelineUrl}
 
 	// Embed additional information into custom extensions
 	cert.ExtraExtensions, err = certificate.Extensions{
-		Issuer: p.issuer,
+		Issuer:                 p.issuer,
+		RunInvocationURI:       jobUrl.String(),
```

Comment:
This is the one I'm most interested in - having the specific job URL in the certificate to provide provenance would be excellent

Reply:
@yob I think it makes sense, we went job-specific for the GitLab provider: https://github.com/sigstore/fulcio/blob/04e4ac9b2d867d0f94e58a858f373a831d87653d/pkg/identity/gitlabcom/principal.go#L233C63-L233C63

Some related discussion in https://github.com/sigstore/fulcio/issues/1182. I found it confusing that this extension was named `RunInvocationURI` (instead of `BuildInvocationURI`), sort of implying it ought to be pipeline-level.

GitHub [does not have `job_id`](https://docs.github.com/en/enterprise-cloud@latest/actions/deployment/security-hardening-your-deployments/about-security-hardening-with-openid-connect#understanding-the-oidc-token) available in their ID tokens, so they don't have the option to align with us on that presently.